indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -295,8 +386,10 @@ class RHRegistrationsExportBase(RHRegistrationsActionBase):
     def _checkParams(self, params):
         RHRegistrationsActionBase._checkParams(self, params)
         reg_list_config = _get_reg_list_config(self.regform)
-        self.items_ids, self.special_items = _get_visible_column_ids(reg_list_config['items'])
-        self.regform_items = RegistrationFormItem.find_all(RegistrationFormItem.id.in_(self.items_ids),
+        item_ids, self.special_item_ids = _split_special_ids(reg_list_config['items'])
+        self.item_ids = _column_ids_to_db(self.regform, item_ids)
+
+        self.regform_items = RegistrationFormItem.find_all(RegistrationFormItem.id.in_(self.item_ids),
```

Comment:
`.find(...).with_parent(self.regform).all()` to restrict to the current registration form would probably make sense